### PR TITLE
[stable18] Document app dev upgrade

### PR DESCRIPTION
**Pull request by ChristophWurst**

### CODE REVIEW

File: developer_manual/app/upgrade-guide.rst [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+* ``Backbone``: ship your own.
+* ``Clipboard``: ship your own.
+* ``ClipboardJs``: ship your own.
+* ``DOMPurify``: ship your own.
+* ``Handlebars``: ship your own.
+* ``jstimezonedetect``: ship your own.
+* ``jstz``: ship your own.
+* ``md5``: ship your own.
+* ``moment``: ship your own.
+* ``OC.basename``: use ``getCurrentUser`` from https://www.npmjs.com/package/@nextcloud/paths
```
Comment (kesselb) [inline developer_manual/app/upgrade-guide.rst:34]: :thinking:

Comment (skjnldsv) [reply developer_manual/app/upgrade-guide.rst:34]: oups :see_no_evil:

**Comment by MorrisJobke** [comment]
```
767 cd build && sh get-server-sources.sh stable18-dev-app-upgrade-guide
768	
769	
770	Fetching source for stable18-dev-app-upgrade-guide
771	
772	Cloning into 'server'...
773	warning: Could not find remote branch stable18-dev-app-upgrade-guide to clone.
774	fatal: Remote branch stable18-dev-app-upgrade-guide not found in upstream origin
775	make: *** [icons-docs] Error 128
776	Makefile:39: recipe for target 'icons-docs' failed
```

🙈 Seems like it cloned because this branch starts with stable18...

**Comment by kesselb** [comment]
Yes. That happens sometimes.

**Pull request merged**